src/generator/generator_visitor.ts

```diff
@@ -230,18 +231,24 @@ class GeneratorVisitor extends Visitor {
 
     protected visitWhileStatement(node: WhileStatement) {
         super.visitWhileStatement(node);
+        this.statements.push(this.createLoopStatement(this.popExpression()));
+    }
 
-        const endLabel = this.generateLabel();
-        const beginLabel = this.generateLabel();
+    protected visitForStatement(node: ForStatement) {
+        super.visitForStatement(node);
 
-        const conditionBranch = this.module.br_if(endLabel, this.module.i32.eqz(this.popExpression()));
-        const loopBranch = this.module.br(beginLabel);
-        const whilePart = this.popStatement();
+        let updateExpression;
+        let condition = this.module.i32.const(1);
 
-        const loopBlock = this.module.block(endLabel, [conditionBranch, whilePart, loopBranch]);
-        const whileStatement = this.module.loop(beginLabel, loopBlock);
+        if (node.update !== null) {
+            updateExpression = this.popStatement();
```

Comment:
Shouldn't this be `this.popExpression()`?

Reply:
No because this is either an `UpdateExpression`, an `AssignmentExpression` or a `ShorthandAssignment`. All of them are `Statements`